src/exceptions.rs

```diff
@@ -1036,6 +1036,7 @@ mod tests {
         use std::error::Error;
 
         Python::with_gil(|py| {
+            #[allow(deprecated)]
```

Comment:
This is the only place I found that is tricky. The problem here is that `Bound<'py, PyBaseException>` doesn't implement `std::error::Error`.

I think the solution will be to update the `impl_exception_boilerplate` macro, but I suspect the naïve approach will run into the same orphan rules problem that I ran into with implementing `std::convert::From` in https://github.com/PyO3/pyo3/pull/3820#discussion_r1492969244. Though https://github.com/PyO3/pyo3/pull/3820#discussion_r1485689848 suggests we shouldn't implement `std::error::Error` here, so I'm not certain what the new way to say this is.